Dockerfile

```diff
@@ -0,0 +1,29 @@
+# Builder
+FROM golang:alpine as builder
+
+WORKDIR /build
+COPY . /build
+
+# Vet and build
+RUN go vet
+RUN go build
```

Comment:
I'd rather we use `make.bash`, as it includes some build info that is very useful when debugging bug reports.

Reply:
... and there is no need to run `go vet` there I think. That is mostly useful to get suggestions on code improvements.

Reply:
> I'd rather we use `make.bash`, as it includes some build info that is very useful when debugging bug reports.

I haven't been able to get this running due to AX.25 dependencies in the Alpine container. It also seems that they don't offer packages for libraries or tools either.

Reply:
The ax.25 dependency should not be needed. Just build with `./make.bash` (without the libax25 sub-command). Should work just fine. If not, please provide the error output so we can debug. Thanks! 🙂

Reply:
Here's the error when building with make.bash:
```
 => ERROR [builder 7/7] RUN sh make.bash                                                                                                                                                             20.3s
------                                                                                                                                                                                                     
 > [builder 7/7] RUN sh make.bash:                                                                                                                                                                         
#14 0.487 WARNING: No static libax25 library available.                                                                                                                                                    
#14 0.487   Linking against shared library instead. To fix                                                                                                                                                 
#14 0.487   this issue, set CGO_LDFLAGS to the full path of
#14 0.487   libax25.a, or run 'make.bash libax25' to download
#14 0.487   and compile libax25-0.0.12-rc4 in .build/
#14 3.490 Downloading Go dependencies...
#14 3.534 Running tests...
#14 14.25 # github.com/la5nta/wl2k-go/transport/ax25
#14 14.25 /go/pkg/mod/github.com/la5nta/wl2k-go@v0.10.0/transport/ax25/ax25.go:143:16: undefined: DialAX25Context
#14 19.17 ?     github.com/la5nta/pat/cfg       [no test files]
#14 19.17 ?     github.com/la5nta/pat/internal/debug    [no test files]
#14 19.17 ?     github.com/la5nta/pat/internal/directories      [no test files]
#14 19.17 ?     github.com/la5nta/pat/internal/forms    [no test files]
#14 19.17 ?     github.com/la5nta/pat/internal/gpsd     [no test files]
#14 19.17 ?     github.com/la5nta/pat/internal/osutil   [no test files]
#14 19.17 ?     github.com/la5nta/pat/internal/buildinfo        [no test files]
#14 19.17 ?     github.com/la5nta/pat/internal/cmsapi   [no test files]
#14 19.17 FAIL  github.com/la5nta/pat [build failed]
#14 19.20 ok    github.com/la5nta/wl2k-go/catalog       0.009s
#14 19.35 ?     github.com/la5nta/wl2k-go/mailbox       [no test files]
#14 19.35 ?     github.com/la5nta/wl2k-go/rigcontrol/hamlib     [no test files]
#14 19.36 ok    github.com/la5nta/wl2k-go/fbb   0.009s
#14 19.66 ok    github.com/la5nta/wl2k-go/lzhuf 0.311s
#14 19.69 ok    github.com/la5nta/wl2k-go/tests 0.023s
#14 19.70 ok    github.com/la5nta/wl2k-go/transport     0.029s
#14 19.84 ok    github.com/la5nta/wl2k-go/transport/ardop       0.003s
#14 20.04 ?     github.com/la5nta/wl2k-go/transport/telnet      [no test files]
#14 20.04 ok    github.com/la5nta/wl2k-go/transport/ardop2      0.004s
#14 20.04 FAIL  github.com/la5nta/wl2k-go/transport/ax25 [build failed]
#14 20.16 ok    github.com/la5nta/wl2k-go/transport/winmor      0.003s
#14 20.16 FAIL
```

Reply:
I'm thinking we don't have the luxury of a reasonably small compiler because if we must build this way we need AX.25 support in the build container which means Alpine is out.

Reply:
I think based on all the tests I've run so far maybe we should leave this Dockerfile as something that just builds our binary and acts as a runtime container. That has the advantage of not requiring heavy build images on SBCs like Raspberry Pis and of working. I think requiring tests to commit new code makes sense, but I'm wondering if committed and tested code needs to be tested again to build a runtime container. What do you think?

Reply:
As a note: I tested with the golang:latest container as the builder and was able to test when running the libax25 install from make.bash, but after building the binary it doesn't end up working in the runtime container because the binary doesn't run. When I swap back to Alpine as the builder without running the tests everything stays happy and works.

Reply:
Ok, I've done some digging. The problem is that golang:alpine does not include [cgo](https://pkg.go.dev/cmd/cgo) support, and this is not handled very well by neither wl2k-go's ax25 code nor make.bash.

I will push a fix to develop that throws a warning when building under these conditions, but still will produce an acceptable build (without libax25). Should be ok for this setup.

Reply:
Done! Pull in the last commit on develop, and `make.bash` should work just fine with golang:alpine (and distroless) 😄

Reply:
> I think requiring tests to commit new code makes sense, but I'm wondering if committed and tested code needs to be tested again to build a runtime container locally. What do you think?

I don't think it's very much overhead to run these tests. Go code compiles and executes very efficiently 🙂 But we have the option to set `ENV SKIP_TESTS=1` in Dockerfile to have make.bash skip them.

Reply:
I set SKIP_TESTS=1 but the build from script still blows up:
```
=> ERROR [builder 7/7] RUN bash ./make.bash                                                                                                                                        14.8s
------                                                                                                                                                                                    
 > [builder 7/7] RUN bash ./make.bash:                                                                                                                                                    
#14 0.198 WARNING: No static libax25 library available.                                                                                                                                   
#14 0.198   Linking against shared library instead. To fix
#14 0.198   this issue, set CGO_LDFLAGS to the full path of
#14 0.198   libax25.a, or run 'make.bash libax25' to download
#14 0.198   and compile libax25-0.0.12-rc4 in .build/
#14 3.199 Downloading Go dependencies...
#14 3.239 Running tests...
#14 3.239 Skipping.
#14 3.239 
#14 3.239 Building Pat v0.13.1...
#14 12.01 # github.com/la5nta/wl2k-go/transport/ax25
#14 12.01 /go/pkg/mod/github.com/la5nta/wl2k-go@v0.10.0/transport/ax25/ax25.go:143:16: undefined: DialAX25Context
------
executor failed running [/bin/sh -c bash ./make.bash]: exit code: 1
```

Reply:
I have it building. You have to disable the tests and prevent the CGO_* vars from getting set in order for it to build in Alpine.

Reply:
Looks like you missed my [recent changes](https://github.com/la5nta/pat/commit/7bac79726643711379c66566cf5aa3bc485471f7)?

In fact, I see that you've created the PR against `master`. Can you please squash and rebase for `develop` HEAD? This should be mentioned in CONTIRBUTING.md. Sorry about that.

My minimal test Dockerfile works now:

```
➜  pat git:(develop) ✗ cat Dockerfile                                
FROM golang:alpine
WORKDIR /src
RUN apk add --no-cache bash perl git
ADD . . 
RUN ./make.bash

➜  pat git:(develop) ✗ docker build -t pat:latest .
#9 [5/5] RUN ./make.bash
#9 sha256:fffcae5c03e843d632e693c80eccc94620ae4a831ea42ef4e424a1d8c55d220f
#9 0.169 WARNING: CGO unavailable. libax25 (ax25+linux) will not be supported with this build.
#9 0.169 Downloading Go dependencies...
#9 6.640 Running tests...
#9 14.65 ?   	github.com/la5nta/pat/cfg	[no test files]
#9 14.65 ?   	github.com/la5nta/pat/internal/buildinfo	[no test files]
#9 14.65 ?   	github.com/la5nta/pat/internal/cmsapi	[no test files]
#9 14.65 ?   	github.com/la5nta/pat/internal/debug	[no test files]
#9 14.65 ?   	github.com/la5nta/pat/internal/directories	[no test files]
#9 14.65 ?   	github.com/la5nta/pat/internal/forms	[no test files]
#9 14.65 ?   	github.com/la5nta/pat/internal/gpsd	[no test files]
#9 14.66 ?   	github.com/la5nta/pat/internal/osutil	[no test files]
#9 14.67 ok  	github.com/la5nta/pat	0.009s
#9 14.68 ok  	github.com/la5nta/wl2k-go/catalog	0.003s
#9 14.69 ?   	github.com/la5nta/wl2k-go/mailbox	[no test files]
#9 14.69 ?   	github.com/la5nta/wl2k-go/rigcontrol/hamlib	[no test files]
#9 14.71 ok  	github.com/la5nta/wl2k-go/fbb	0.014s
#9 14.94 ok  	github.com/la5nta/wl2k-go/lzhuf	0.245s
#9 14.94 ok  	github.com/la5nta/wl2k-go/tests	0.018s
#9 14.94 ok  	github.com/la5nta/wl2k-go/transport	0.016s
#9 15.01 ok  	github.com/la5nta/wl2k-go/transport/ardop	0.009s
#9 15.05 ok  	github.com/la5nta/wl2k-go/transport/ardop2	0.003s
#9 15.06 ok  	github.com/la5nta/wl2k-go/transport/ax25	0.003s
#9 15.18 ?   	github.com/la5nta/wl2k-go/transport/telnet	[no test files]
#9 15.18 ok  	github.com/la5nta/wl2k-go/transport/ax25/agwpe	0.003s
#9 15.22 ok  	github.com/la5nta/wl2k-go/transport/winmor	0.003s
#9 15.30 
#9 15.30 Building Pat v0.13.1...
#9 16.06 Enjoy!
#9 DONE 16.3s
```